### ISSUE TITLE
Vehicle Save Error

### ISSUE THREAD

**Issue by Pierrive** [Vdauphin/HeartsAndMinds#992]
Hello,

Our team has been playing on your mission for more than a year now. Mission that I took the liberty of modifying by adding some personal script. You will find it on: https://github.com/Pierrive/HeartsAndMinds

Latest current version: =BTC=co@30_Hearts_and_Minds_New.Farkhar
Mods Use : https://team-vlr.fr/jouer-avec-nous/

My question:
- Have you had any errors reported on the backup of modded vehicles, especially concerning RHS vehicles?

Indeed, it sometimes happens that after a backup, all the vehicles listed Btc_vehicle disappears, and this without generating an error in the RPT. I had to take out the function to save the vehicles from your save.sqf file, to put it aside, so that it does not impact the other variable backups. 

I also integrated an autocorrection in case this event occurs, but I find it a bit too drastic.

If you have a little time to answer me and / or have a look at what's going on, I'd appreciate it.

PS: A huge bravo for the correction work you continue to do on this wonderful mission.

**Comment by Vdauphin** [comment]
Hello,

> Have you had any errors reported on the backup of modded vehicles, especially concerning RHS vehicles?

No.

> all the vehicles listed Btc_vehicle disappears

When loading vehicle from database the mission remove all vehicles from the mission.sqm (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L116). Then it load vehicle data previously saved from the `profileNamespace` (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L119).
Finally, vehicles are created (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L124-L159).

> I had to take out the function to save the vehicles from your save.sqf file, to put it aside, so that it does not impact the other variable backups.

So if you remove this https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/save.sqf#L137-L163 the database will not load vehicles.

> PS: A huge bravo for the correction work you continue to do on this wonderful mission.

Happy you guys still enjoy it!

Cheers.

**Comment by Pierrive** [comment]
Thank you for your prompt response. Indeed, I expressed myself badly. Unfortunately, I went through a translator, to try to dialogue in English.

I didn't remove the backup function, I simply removed it from the continuity of the backup script.

https://github.com/Pierrive/HeartsAndMinds/blob/master/=BTC=co@30_Hearts_and_Minds_New.Farkhar/core/fnc/db/save.sqf#L160-L177

https://github.com/Pierrive/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds_New.Farkhar/core/fnc/db/saveVeh.sqf

Then in the fnc saveVeh, I added a patch : 
https://github.com/Pierrive/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds_New.Farkhar/core/fnc/db/saveVeh.sqf#L35-L68

Sorry for my English which is catastrophic.

Enjoy

**Comment by Vdauphin** [comment]
> Thank you for your prompt response. Indeed, I expressed myself badly. Unfortunately, I went through a translator, to try to dialogue in English.

No problem.

First make sure your changes work.
Load your mission locally, move a vehicle randomly then save the mission by executing: `[] spawn btc_fnc_db_save;` in debug console (exec server). When it is done, just quit the mission and enter in it again. 
If your changes work, you should see the vehicle you moved previously.

**Comment by GoldJohnKing** [comment]
If you @Pierrive are talking about `btc_veh`s disappear at next start which it's loaded from a previous save, I may have encountered the same problem.

Unfortunately, I'm running a heavily modified version of H&M, too. Though the issue appears quite often, it still happens randomly and I still didn't come up with an idea to reproduce it for certain, so I never reported it here. As someone might have the same issue as I have... It feels like home... 🤣🤣🤣

Among those edited contents, which may be related to the database are, the cleanUp function which will delete all opfor units far from players (3000m) on triggering, and I've appended many vehicles and AI handling weapons (like UGV) in the logistics menu.

I wonder if there's something which would cause either save or load malfunction under edge conditions.

I'd like to know how to get all the saved arrays (cities, enemies, `btc_veh`s) so I may help my self solving the issue. Besides, I would kindly ask for a debug function on save and load functions which would logged into rpt directly, so we can get see what's going wrong. I think this is much better than going through what we have modified on the original H&M.
